jetty-security/src/main/java/org/eclipse/jetty/security/authentication/AuthorizationService.java

```diff
@@ -0,0 +1,49 @@
+//
+//  ========================================================================
+//  Copyright (c) 1995-2018 Mort Bay Consulting Pty. Ltd.
+//  ------------------------------------------------------------------------
+//  All rights reserved. This program and the accompanying materials
+//  are made available under the terms of the Eclipse Public License v1.0
+//  and Apache License v2.0 which accompanies this distribution.
+//
+//      The Eclipse Public License is available at
+//      http://www.eclipse.org/legal/epl-v10.html
+//
+//      The Apache License v2.0 is available at
+//      http://www.opensource.org/licenses/apache2.0.php
+//
+//  You may elect to redistribute this code under either of these licenses.
+//  ========================================================================
+//
+
+package org.eclipse.jetty.security.authentication;
+
+import javax.servlet.http.HttpServletRequest;
+
+import org.eclipse.jetty.security.LoginService;
+import org.eclipse.jetty.server.UserIdentity;
+
+/**
+ * <p>A service to query for user roles.</p>
+ */
+public interface AuthorizationService
+{
+    /**
+     * @param request the current HTTP request
+     * @param name the user name
+     * @param credentials the user credentials
+     * @return a {@link UserIdentity} to query for roles of the given user
+     */
+    UserIdentity getUserIdentity(HttpServletRequest request, String name, Object credentials);
+
+    /**
+     * <p>Wraps a {@link LoginService} as an AuthorizationService</p>
+     *
+     * @param loginService the {@link LoginService} to wrap
+     * @return an AuthorizationService that delegates the query for roles to the given {@link LoginService}
+     */
+    public static AuthorizationService from(LoginService loginService)
+    {
+        return (request, name, credentials) -> loginService.login(name, credentials, request);
```

Comment:
If you are not passing in a fixed credential here, then I see little point in AuthorizationService?   You are just changing `login(name,cred,req)` to `getUserIdentity(req,name,cred)`?

At the very least, keep the argument order the same, but it makes more sense to me to pass in a fixed credential here rather than pass in "" when calling.  What if a particular loginService can't handle empty credentials?  It would be better to be able to configure "password" here rather than have "" baked into the caller.